### PR TITLE
clang++-inspired cleanups

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -406,10 +406,8 @@ int32_t Copter::Mode::get_alt_above_ground(void)
 void Copter::Mode::land_run_vertical_control(bool pause_descent)
 {
 #if PRECISION_LANDING == ENABLED
-    AC_PrecLand &precland = copter.precland;
-
     const bool navigating = pos_control->is_active_xy();
-    bool doing_precision_landing = !ap.land_repo_active && precland.target_acquired() && navigating;
+    bool doing_precision_landing = !ap.land_repo_active && copter.precland.target_acquired() && navigating;
```
Comment (OXINARF) [inline ArduCopter/mode.cpp]: Is the `copter` part needed? I would think yes, but then I don't see how shadowing was happening.

File: ArduCopter/mode_auto.cpp [hunk 2]
```diff
@@ -1126,28 +1132,7 @@ void Copter::ModeAuto::do_loiter_unlimited(const AP_Mission::Mission_Command& cm
...
-    // default target altitude to current altitude if not provided
-    if (circle_center.alt == 0) {
-        int32_t curr_alt;
-        if (current_loc.get_alt_cm(circle_center.get_alt_frame(),curr_alt)) {
-            // circle altitude uses frame from command
-            circle_center.set_alt_cm(curr_alt,circle_center.get_alt_frame());
-        } else {
-            // default to current altitude above origin
-            circle_center.set_alt_cm(current_loc.alt, current_loc.get_alt_frame());
-            copter.Log_Write_Error(ERROR_SUBSYSTEM_TERRAIN, ERROR_CODE_MISSING_TERRAIN_DATA);
```
Comment (OXINARF) [inline ArduCopter/mode_auto.cpp]: I don't think this code was correct, but just pointing out this is missing now.

File: ArduCopter/mode_acro.cpp [hunk 3]
```diff
@@ -118,16 +116,17 @@ void Copter::ModeAcro::get_pilot_desired_angle_rates(int16_t roll_in, int16_t pi
...
         // Calculate angle limiting earth frame rate commands
         if (g.acro_trainer == ACRO_TRAINER_LIMITED) {
-            if (roll_angle > aparm.angle_max){
-                rate_ef_level.x -=  g.acro_balance_roll*(roll_angle-aparm.angle_max);
-            }else if (roll_angle < -aparm.angle_max) {
-                rate_ef_level.x -=  g.acro_balance_roll*(roll_angle+aparm.angle_max);
+            const float angle_max = copter.aparm.angle_max;
+            if (roll_angle > angle_max){
+                rate_ef_level.x -=  g.acro_balance_roll*(roll_angle-angle_max);
+            }else if (roll_angle < -angle_max) {
```
Comment (khancyr) [inline ArduCopter/mode_acro.cpp]: you could have correct the style on those one ;-P

File: libraries/AP_Gripper/AP_Gripper_Servo.cpp [hunk 4]
```diff
@@ -49,23 +49,18 @@ bool AP_Gripper_Servo::has_state_pwm(const uint16_t pwm) const
...
-    }
-    return false;
-#else
     if (AP_HAL::millis() - action_timestamp < action_time) {
         // servo still moving....
         return false;
     }
-    return true;
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (is_releasing) {
```
Comment (khancyr) [inline libraries/AP_Gripper/AP_Gripper_Servo.cpp]: hum ok for this regression.

**Pull request merged**